src/types.js

```diff
@@ -59,9 +63,70 @@ export type SpacingAliases = {
   +[spacingAlias: string]: number,
 }
 
+export type DisplayAliases = {
+  +[displayAlias: string]: string,
+}
+
 export type ConfigProviderContext = {|
   +xnReflex: {|
+    +displayAliases: DisplayAliases,
     +spacingAliases: SpacingAliases,
     +base: number,
   |},
 |}
+
+export type OneResolutionGrid = {
+  align?: AlignGrid,
+  children?: React.Node,
+  className?: string,
+  dev?: Dev,
+  justify?: Justify,
+  margin?: Spacing,
+  marginBottom?: SpacingValues,
+  marginLeft?: SpacingValues,
+  marginRight?: SpacingValues,
+  marginTop?: SpacingValues,
+  padding?: Spacing,
+  paddingBottom?: SpacingValues,
+  paddingLeft?: SpacingValues,
+  paddingRight?: SpacingValues,
+  paddingTop?: SpacingValues,
+  show?: DisplayValues,
+  size?: Size,
+  style?: { [string]: string | number },
+}
+
+export const resolutionProperties = [
+  'align',
+  'justify',
+  'margin',
+  'marginBottom',
+  'marginLeft',
+  'marginRight',
+  'marginTop',
+  'padding',
+  'paddingBottom',
+  'paddingLeft',
+  'paddingRight',
+  'paddingTop',
+  'size',
+]
+
+export type GridProps = {
+  ...OneResolutionGrid,
+  align?: AlignGrid | { [string]: AlignGrid },
+  justify?: Justify | { [string]: Justify },
+  margin?: Spacing | { [string]: Spacing },
+  marginBottom?: SpacingValues | { [string]: SpacingValues },
```

Comment:
I don't think these flowtypes are correct. https://flow.org/en/docs/types/objects/#toc-objects-as-maps

`{[string]: SpacingValues}` tells me that I can pass in this to marginBottom:
```
marginBottom={foo: 1}
```

Reply:
I had noticed this too. The issue is that the possible keys are dynamic based on aliases. It's possible we can address this with generic typing. But I'm ok with a little less specific typing and addressing in a future PR.

Reply:
Not sure I follow.  With aliases this could be something like `marginBottom?: string | $Keys<typeof spacingAliases>` -> `marginBottom="XL"`

Reply:
With it you can now use:

```
<Grid marginBottom={{ large: 'M', medium 'S', small: 'XS' }} />
```

In addition to being able to say:

```
<Grid show="large" marginBottom="M" />
<Grid show="medium" marginBottom="S" />
<Grid show="small" marginBottom="XS" />
```

The difference is that with the first approach you maintain the same component so state is preserved. It won't be always possible to use the first approach which is why the second one is there.

The reason why the types are defined as `[string]` instead of `"large" | "medium" | "small"` is because these are just the defaults